ingestion/src/metadata/ingestion/api/topology_runner.py

```diff
@@ -159,22 +159,30 @@ def sink_request(self, stage: NodeStage, entity_request: C) -> Iterable[Entity]:
 
         if entity is not None:
             if stage.ack_sink:
-                tries = 3
                 entity = None
 
                 entity_fqn = self.fqn_from_context(
                     stage=stage, entity_request=entity_request
                 )
 
-                while not entity and tries > 0:
-                    yield entity_request
-                    # Improve validation logic
+                if not stage.overwrite:
```

Comment:
Could we pls add a comment here explaining a bit the behavior of the overwrite?

Reply:
this function keeps getting a bit more complicated each time so either adding more comments here or breaking it down at some point might make sense 🤔

Reply:
I agree it would make sense but this PR has already few changes. What do you think about doing that in a different task?